### PR TITLE
vscode-extensions.QURI.vscode-squiggle: init at 0.5.0

### PR DESCRIPTION
**Pull request by quinn-dougherty**
###### Description of changes

https://squiggle-language.com 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by quinn-dougherty** [comment]
@jonringer are you the point person for vscode extensions? I followed the templates from the other extensions, is this gonna work?

File: pkgs/applications/editors/vscode/extensions/squiggle/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ lib
+, vscode-utils
+}:
+let
+  version = "0.5.0";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/squiggle/default.nix:5]:
```suggestion
  version = "0.5.0";
```
Is version re-used somewhere? If not, keep in scope.

File: pkgs/applications/editors/vscode/extensions/squiggle/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    inherit version;
+    sha256 = "sha256-xkl6K2F3lc8bCds4pz/tHRiuAyk8l5HRfp9nxkUj/AI=";
+  };
+
+  meta = with lib; {
+    description = "A simple programming language for intuitive probabilistic estimation";
+    downloadPage = "https://marketplace.visualstudio.com/items?itemName=QURI.vscode-squiggle";
+    homepage = "https://squiggle-language.com";
+    license = licenses.mit;
+    platforms = [ "x86_64-linux" "aarch64-darwin" "x86_64-darwin" ];
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/squiggle/default.nix:19]:
```suggestion
    platforms = [ "x86_64-linux" "aarch64-darwin" "x86_64-darwin" ];
```
Why not `platforms.unix`?

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 3]
```diff
@@ -2047,6 +2047,8 @@ let
         };
       };
 
+      QURI.vscode-squiggle = callPackage ./squiggle { };
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:2050]:
```suggestion
      QURI.vscode-squiggle = callPackage ./squiggle { };
```
Upstream uses name as: `QURI.vscode-squiggle`
Ideally, we should follow up with that.

**Comment by superherointj** [comment]
> @jonringer are you the point person for vscode extensions?

Jon is on sabbatical. Sorry for the long delay in this PR. I'll handle this.

> I followed the templates from the other extensions, is this gonna work?

Yes, you did well. Were you able to make it work for yourself?

I will have this extension merged, just follow up my feedback. So we can get this done.

**Comment by superherointj** [comment]
Please fix merge conflict.

**Comment by superherointj** [comment]
ping @quinn-dougherty

**Comment by superherointj** [comment]
ping @quinn-dougherty

File: pkgs/applications/editors/vscode/extensions/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/applications/editors/vscode/extensions/default.nix:2050]:
```suggestion
      quri.vscode-squiggle = callPackage ./quri.vscode-squiggle { };
```
we lowercase things

**Comment by drupol** [comment]
Closing the PR, feel free to reopen it if you plan to update it.

**Pull request closed without merging**